src/include/iscsi-lio-server/TargetData.rb

```diff
@@ -0,0 +1,594 @@
+        @cmd_out = `#{@cmd}`
+        @current_acl_rule.store_password(@cmd_out[9 , @cmd.length])
+        @cmd = "targetcli iscsi/" + @current_target.fetch_target_name() + \
+            "/tpg" + @current_tpg.fetch_tpg_number() + "/acls/" + @initiator_name + "/ get auth mutual_userid"
+        @cmd_out = `#{@cmd}`
+        @current_acl_rule.store_mutual_userid(@cmd_out[14 , @cmd.length])
+        @cmd = "targetcli iscsi/" + @current_target.fetch_target_name() + \
+            "/tpg" + @current_tpg.fetch_tpg_number() + "/acls/" + @initiator_name + "/ get auth mutual_password"
+        @cmd_out = `#{@cmd}`
+        @current_acl_rule.store_mutual_password(@cmd_out[16 , @cmd.length])
+      end
+
+      # handle mapped luns here
+      if @re_mapped_lun_line.match(line)
+        @mapping_lun_name = @re_mapping_lun.match(line).to_s.strip
+        @mapped_lun_name = @re_mapped_lun.match(line).to_s.strip
+        @mapped_lun_name.slice!("[")
+        mapping_lun_num = @mapping_lun_name[10,@mapping_lun_name.length]
+        @current_acl_rule.store_mapped_lun(mapping_lun_num)
+        mapped_lun_num = @mapped_lun_name[3,@mapped_lun_name.length]
+        @current_acl_rule.fetch_mapped_lun(mapping_lun_num).store_mapped_lun_number(mapped_lun_num)
+      end
+
+      # handle luns here
+      if @re_lun.match(line)
+        # lun_num is a string like lun0, lun1,lun2....
+        lun_num_tmp = @re_lun_num.match(line).to_s
+        lun_num = lun_num_tmp[2,lun_num_tmp.length]
+        lun_name_tmp = line[line.index("[")+1 .. line.index("(")-2]
+        lun_name = lun_name_tmp[lun_name_tmp.index("/")+1 .. lun_name_tmp.length]
+        # lun_num_int is a number like 1,3,57.
+        lun_num_int = lun_num[3,lun_num.length]
+        lun_path_tmp = @re_lun_path.match(line).to_s
+        lun_path = lun_path_tmp[1,lun_path_tmp.length-2]
+        @current_tpg.store_lun(lun_num,[rand(9999), lun_num_int, lun_name, lun_path, File.ftype(lun_path)])
+      end
+
+      # handle portals here
+      if @re_portal.match(line)
+        portal_line = @re_portal.match(line).to_s
+        index = portal_line.index(":")
+        ip = portal_line[0,index]
+        port = portal_line[index+1, portal_line.length]
+        # Yast only support one ip port pair now
+        @current_tpg.store_portal(ip, port)
+      end
+
+    end # end of @target_outout.each do |line|
+
+  end # end of the function
+
+  def print_targets()
+    @targets_list.print_list()
+  end
+
+ # this function will return are created target names.
+  def get_target_names_array()
+    names = Array.new
+    names = @targets_list.get_target_names()
+    return names
+  end
+
+  # This function will return the Hash list target_list
+  def get_target_list()
+    list = @targets_list
+    return list
+  end
+end
+
+class DiscoveryAuth
+  def initialize
+    @discovery_auth = Hash.new()
+  end
+
+  def store_status(status)
+    @discovery_auth.store("status", status)
```

Comment:
`somehash.store(key, value)` is just a longer way to write `somehash[key] = value`

BTW not so for the getter,
`somehash.fetch(key)` may raise KeyError where `somehash[key]` would return `nil` if `key` is not there.